sway-core/src/language/ty/expression/expression_variant.rs

```diff
@@ -51,7 +52,7 @@ pub enum TyExpressionVariant {
         struct_name: Ident,
         fields: Vec<TyStructExpressionField>,
         span: Span,
-        type_binding: TypeBinding<()>,
+        call_path_binding: TypeBinding<CallPath>,
```

Comment:
What is the difference between `call_path_binding` and `struct_name`? And could these two be combined?

Reply:
The struct_name is obtained from the `TypeInfo` so I think it's actually the definition identifier, or at least supposed to be. This is different from the local `call_path_binding`, whose suffix could be similar but not necessarily.

I don't know how exactly it would behave if you had type aliases, for instance.

In any case I think it makes sense to hold both as it is.